sdks/java/io/rrio/src/test/java/org/apache/beam/io/requestresponse/RequestResponseIOTest.java

```diff
@@ -0,0 +1,443 @@
+    visitor.assertNotExpandsWith(CACHE_WRITE_NAME);
+  }
+
+  @Test
+  public void givenCacheUsingRedis_transformExpandsWithCallAndCache()
+      throws NonDeterministicException {
+    ExpansionPipelineVisitor visitor = new ExpansionPipelineVisitor();
+    requests()
+        .apply(
+            RequestResponseIO.of(new CallerImpl(), RESPONSE_CODER)
+                .withRedisCache(
+                    URI.create("redis://localhost:6379"),
+                    REQUEST_CODER,
+                    Duration.standardHours(1L)));
+
+    pipeline.traverseTopologically(visitor);
+
+    Call.Configuration<?, ?> configuration =
+        visitor.assertExpandsWithCallOf(CALL_NAME, WRAPPED_CALLER, NOOP_SETUP_TEARDOWN);
+    assertThat(configuration.getTimeout(), equalTo(DEFAULT_TIMEOUT));
+    assertThat(configuration.getShouldRepeat(), equalTo(true));
+    assertThat(
+        configuration.getBackOffSupplier().getClass(),
+        equalTo(DefaultSerializableBackoffSupplier.class));
+    assertThat(
+        configuration.getSleeperSupplier().get().getClass(), equalTo(Sleeper.DEFAULT.getClass()));
+    assertThat(
+        configuration.getCallShouldBackoff().getClass().getName(),
+        equalTo(WRAPPED_CALL_SHOULD_BACKOFF));
+
+    visitor.assertNotExpandsWith(THROTTLE_NAME);
+    visitor.assertExpandsWithCallOf(
+        CACHE_READ_NAME, CACHE_READ_USING_REDIS, CACHE_READ_USING_REDIS);
+    visitor.assertExpandsWithCallOf(
+        CACHE_WRITE_NAME, CACHE_WRITE_USING_REDIS, CACHE_WRITE_USING_REDIS);
+  }
+
+  @Test
+  public void givenAllConfigurationUsingRedis_transformExpandsWithEverything()
+      throws NonDeterministicException {
+    ExpansionPipelineVisitor visitor = new ExpansionPipelineVisitor();
+    requests()
+        .apply(
+            RequestResponseIO.of(new CallerImpl(), RESPONSE_CODER)
+                .withRedisCache(
+                    URI.create("redis://localhost:6379"),
+                    REQUEST_CODER,
+                    Duration.standardHours(1L)));
+
+    pipeline.traverseTopologically(visitor);
+
+    Call.Configuration<?, ?> configuration =
+        visitor.assertExpandsWithCallOf(CALL_NAME, WRAPPED_CALLER, NOOP_SETUP_TEARDOWN);
+    assertThat(configuration.getTimeout(), equalTo(DEFAULT_TIMEOUT));
+    assertThat(configuration.getShouldRepeat(), equalTo(true));
+    assertThat(
+        configuration.getBackOffSupplier().getClass(),
+        equalTo(DefaultSerializableBackoffSupplier.class));
+    assertThat(
+        configuration.getSleeperSupplier().get().getClass(), equalTo(Sleeper.DEFAULT.getClass()));
+    assertThat(
+        configuration.getCallShouldBackoff().getClass().getName(),
+        equalTo(WRAPPED_CALL_SHOULD_BACKOFF));
+
+    visitor.assertExpandsWithCallOf(
+        CACHE_READ_NAME, CACHE_READ_USING_REDIS, CACHE_READ_USING_REDIS);
+    visitor.assertExpandsWithCallOf(
+        CACHE_WRITE_NAME, CACHE_WRITE_USING_REDIS, CACHE_WRITE_USING_REDIS);
+  }
```

Comment:
@johnjcasey or @Abacn Thank you.

Reply:
I would be careful with this category of tests. Particular expansion checks can be fragile, so I would expect these tests to be flakier than an 'average' test. Also, while these verify configuration is set, they don't actually execute the pipeline. 

I would look to add a few tests with a mock call implemented, so as to verify behavior in both happy path and error scenarios.